frontend/ui/widget/textviewer.lua

```diff
@@ -531,6 +542,93 @@ function TextViewer:handleTextSelection(text, hold_duration, start_idx, end_idx,
     end
 end
 
+function TextViewer:reinit()
+    local low, high = self.scroll_text_w.text_widget:getVisibleHeightRatios() -- try to keep position
+    local ratio = low == 0 and 0 or (low + high) / 2 -- if we are at the beginning, keep the first line visible
+    UIManager:setDirty(nil, "partial", self.frame.dimen)
```

Comment:
That's to clear the "old"/"current" instance in case the new one has different dimensions, right?

(No further comments, just wanted to confirm, as it's the exact right way to do that sort of thing ^^).

Reply:
Yes, as I wrote above:
The latest commit adds clearing the TextViewer area when reinitting it after changing the monospace font.
With non-monospace font the TextViewer window is taller than with monospace font (depending on the font size, for example by 3 pixels for font size 20 on my Kindle).